include/ck_queue.h

```diff
@@ -153,6 +153,11 @@ struct {									\
 	    (var);								\
 	    (var) = CK_SLIST_NEXT((var), field))
 
+#define	CK_SLIST_FOREACH_FROM(var, head, field)					\
+	for ((var) = (var) != NULL ? (var) : CK_SLIST_FIRST((head));		\
```

Comment:
I'd feel a bit more comfortable if there were extra parenthesis here, as is done with FreeBSD, ie (var) = ((var) != NULL ? (var) : CK_SLIST_FIRST((head)))

Reply:
Done!